Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -234,6 +247,118 @@ Analytics payload
             return WeaveModuleResult.Success(weaveResults);
         }
 
+        private WeaveTypeResult WeaveGeneratedType(TypeDefinition type)
+        {
+            _logger.Debug("Weaving generated " + type.Name);
+
+            var interfaceName = $"I{type.Name}Accessor";
+            var interfaceType = _moduleDefinition.GetType($"{type.Namespace}.Generated", interfaceName);
+
+            var persistedProperties = new List<WeavePropertyResult>();
+
+            // We need to weave all (and only) the properties in the accessor interface
+            foreach (var interfaceProperty in interfaceType.Properties)
+            {
+                var prop = type.Properties.First(p => p.Name == interfaceProperty.Name);
+
+                try
+                {
+                    var weaveResult = WeaveGeneratedClassProperty(type, prop, interfaceType);
+                    persistedProperties.Add(weaveResult);
+                }
+                catch (Exception e)
+                {
+                    var sequencePoint = prop.GetSequencePoint();
+                    _logger.Error(
+                        $"Unexpected error caught weaving property '{type.Name}.{prop.Name}': {e.Message}.\r\nCallstack:\r\n{e.StackTrace}",
+                        sequencePoint);
+
+                    return WeaveTypeResult.Error(type.Name, isGenerated: true);
+                }
+            }
+
+            return WeaveTypeResult.Success(type.Name, persistedProperties, isGenerated: true);
+        }
+
+        private WeavePropertyResult WeaveGeneratedClassProperty(TypeDefinition type, PropertyDefinition prop, TypeDefinition interfaceType)
+        {
+            var accessorGetter = new MethodReference($"get_Accessor", interfaceType, type) { HasThis = true };
+
+            ReplaceGeneratedClassGetter(prop, interfaceType, accessorGetter);
+
+            if (prop.SetMethod != null)
+            {
+                ReplaceGeneratedClassSetter(prop, interfaceType, accessorGetter);
+            }
+
+            return WeavePropertyResult.Success(prop);
+        }
+
+        private void ReplaceGeneratedClassGetter(PropertyDefinition prop, TypeDefinition interfaceType, MethodReference accessorGetter)
```

Comment:
```suggestion
        private static void ReplaceGeneratedClassGetter(PropertyDefinition prop, TypeDefinition interfaceType, MethodReference accessorGetter)
```

Reply:
Already done in the other PR (I will merge back the changes in Realm and Realm.SourceGenerator afterI go through the comments)